wireapp/wire-ios-mono
# fix: request loop slow sync - WPB-6502

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6502" title="WPB-6502" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-6502</a>  [iOS] request loop on /notifications and app stuck in sync
  </td></table>
  <br />
 




















----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

During the 3.111.X releases, a lot of users had database crashes and resulted in clearing the database. Doing that they will start fresh but with a persisted lastEventId in the userDefaults. So we'll not get a correct lastEventId for their new client.

### Causes (Optional)

We started the slow sync with the wrong case: fetchingMissingEvents (`/notifications) with an incorrect lastEventId.

if /notifications returns a 404, we'll force a slowSync and clearing the lastEventId. Unfortunately this forceSlowSync was starting on the wrong phase `.fetchingMissingEvents` instead of `.fetchingLastUpdateEventID` (`/notifications/last`). So we'll end up doing again (request loop) a `/notifications` returning a 404, and triggering a slowSync.

### Solutions

Start the slowSync with `.fetchingLastUpdateEventID`

In some cases, we have (old) database patches where a slowSync was triggered in order to sync resources, this behavior has been kept and moved to own method.
### Testing

#### Test Coverage (Optional)


#### How to Test

* Insert duplicate user from previous build
* Upgrade, this will trigger a slow sync after the migration
There should not be a request loop

### Notes (Optional)

The database patches could be removed, but this can be done in a second time to clean up.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
